core/testing/environment.go

```diff
@@ -0,0 +1,68 @@
+package coretesting
+
+import (
+	"context"
+
+	appmodulev2 "cosmossdk.io/core/appmodule/v2"
+	corecontext "cosmossdk.io/core/context"
+	corelog "cosmossdk.io/core/log"
+	"cosmossdk.io/core/router"
+	"cosmossdk.io/core/store"
+)
+
+type TestEnvironmentConfig struct {
+	ModuleName  string
+	Logger      corelog.Logger
+	MsgRouter   router.Service
+	QueryRouter router.Service
+}
+
+type TestEnvironment struct {
+	env              appmodulev2.Environment
+	memEventsService MemEventsService
```

Comment:
Include the concrete types of our `Mem` services so that we can use the additional helper methods (such as `GetEvents`) in our tests